### PR TITLE
AWS Parameter Store guide

### LINKED ISSUE #991: Move aws-parameter-store-annotation example to guide
Currently the sample app for aws-parameter-store-annotation` does not test any functionality. I am going to move it to `micronaut-guides-poc` repo.

### PR DESCRIPTION
**Pull request by sdelamo**
See: https://github.com/micronaut-projects/micronaut-starter/pull/702

Close: https://github.com/micronaut-projects/micronaut-aws/issues/991

To generate the guide:

`./gradlew -Dmicronaut.guide=micronaut-aws-parameter-store build`

To open the HTML 

`open build/dist/micronaut-aws-parameter-store-gradle-java.html`

### CODE REVIEW

**Comment by sdelamo** [comment]
@pgressa I have changed the example from the sample app to something which I think it i easier to understand with environments. Let me know what you think.

**Review by ilopmar — APPROVED** [review]
Looks good. I've pushed a few commits with some minor things.

As you are using a snapshot micronaut version because of the param-store feature, we should wait and do not merge this until Micronaut 2.4.2 is out.

**Comment by sdelamo** [comment]
> As you are using a snapshot micronaut version because of the param-store feature, we should wait and do not merge this until Micronaut 2.4.2 is out.

Yes, that it is exactly the reason.

**Comment by ilopmar** [comment]
Please change the PR to Draft to make sure we don't accidentally merge it.

**Review by pgressa — APPROVED** [review]
@sdelamo yeah no problem, looks good! Thanks for taking care of this.

**Comment by pgressa** [comment]
@sdelamo should I also test it or code review is enough ?

File: guides/micronaut-aws-parameter-store/micronaut-aws-parameter-store.adoc [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+dependency:micronaut-aws-parameter-store[groupId=io.micronaut.aws]
+
+=== Bootstrap.yml
+
+To enable distributed configuration a src/main/resources/bootstrap.yml configuration file must be created and configured to use Parameter Store:
+
+resource:bootstrap.yml[]
+
+`micronaut.application.name` name will be part of the parameter name in AWS Parameter Store.
+With `aws.client.system-manager.parameterstore.enabled` enable distributed configuration with AWS Parameter Store.
```
Comment (pgressa) [inline guides/micronaut-aws-parameter-store/micronaut-aws-parameter-store.adoc]: I think this is missing here:
```
micronaut:
  config-client:
    enabled: true
```

Comment (sdelamo) [reply guides/micronaut-aws-parameter-store/micronaut-aws-parameter-store.adoc]: Good catch. The feature generates [micronaut.config-client.enabled]( https://github.com/micronaut-projects/micronaut-starter/blob/2.4.x/starter-core/src/main/java/io/micronaut/starter/feature/awsparameterstore/AwsParameterStore.java#L49) but that configuration does not appear in our [configuration documentation](https://docs.micronaut.io/latest/guide/configurationreference.html#io.micronaut.runtime.ApplicationConfiguration). I am going to look in the javadoc to see if I find what it does.

Comment (sdelamo) [reply guides/micronaut-aws-parameter-store/micronaut-aws-parameter-store.adoc]: I've added a sentence to the guide about `micronaut.config-client.enabled`. Moreover, the javadoc seemed to be wrong as well. https://github.com/micronaut-projects/micronaut-core/pull/5178

**Comment by sdelamo** [comment]
> @sdelamo should I also test it or code review is enough ?

It is ok. I tested against my AWS account.

**Comment by sdelamo** [comment]
@ilopmar  I changed this to use 2.4.2 instead of snapshot now that 2.4.2 has been released. 

If the test pass we can merge it.

**Comment by ilopmar** [comment]
@sdelamo sounds good. I've already upgraded the guides to 2.4.2: https://github.com/micronaut-projects/micronaut-guides-poc/pull/174

**Pull request merged**